luaui/Widgets/cmd_customformations2.lua

```diff
@@ -1337,10 +1341,20 @@ function stepFiveStar(colcover, rowcover, row, col, n, starscol, primescol)
     end
 end
 
+
 function widget:Initialize()
 	WG.customformations_linelength = 0
+
+	WG.customformations = {}
+	WG.customformations.getRepeatForSingleUnit = function()
+		return repeatForSingleUnit
+	end
+	WG.customformations.setRepeatForSingleUnit = function(value)
+		repeatForSingleUnit = value
+	end
 end
 
+
 function widget:Shutdown()
-	WG.customformations_linelength = 0
+	WG.customformations = nil
```

Comment:
I think you still need the `WG.customformations_linelength = 0`, since for some reason it's a separate field on `WG` than `WG.customformations`. It looks like that field is only used in this file, so maybe it can be switched to `WG.customformations.linelength` instead?

Reply:
Oh woops, I was being sloppy there, nice catch. Does indeed look like it's only used here, so I'll just make it local.